lisa/tools/sysbench.py

```diff
@@ -0,0 +1,561 @@
+                f" --file-merged-requests={file_merged_requests}"
+                f" --file-rw-ratio={file_rw_ratio}"
+            )
+
+            prepare_cmd: str = (
+                f"fileio prepare --file-total-size={total_size} --file-num={total_file}"
+            )
+            self.run(
+                prepare_cmd,
+                force_run=True,
+            )
+
+            res = self.__run_sysbench_test(
+                test_type="fileio",
+                threads=threads,
+                events=events,
+                time_limit=time_limit,
+                validation=False,
+                extra_args=extra_args,
+            )
+
+            cleanup_cmd: str = (
+                f"fileio cleanup --file-total-size={total_size} --file-num={total_file}"
+            )
+            self.run(
+                cleanup_cmd,
+                force_run=True,
+            )
+
+            perf_msg = self.__process_fileio_perf_result(
+                res.stdout,
+                operation,
+            )
+            perf_msg["file_fsync_all"] = file_fsync_all
+            perf_msg["file_fsync_end"] = file_fsync_end
+            perf_msg["total_file"] = total_file
+            perf_msg["file_total_size_in_gb"] = file_total_size_in_gb
+            perf_msg["file_async_backlog"] = file_async_backlog
+            perf_msg["file_fsync_freq"] = file_fsync_freq
+            perf_msg["file_merged_requests"] = file_merged_requests
+            perf_msg["file_rw_ratio"] = file_rw_ratio
+            perf_msg["file_ops"] = operation
+            perf_msg["file_io_mode"] = file_io_mode
+            perf_msg["file_fsync_mode"] = file_fsync_mode
+
+            self.__send_subtest_msg(
+                message_type=DiskPerformanceMessage,
+                test_name=testcase_name,
+                test_result=test_result,
+                other_fields=perf_msg,
+            )
+
+    def run_memory_perf(
+        self,
+        test_result: Any,
+        threads: int = 1,
+        events: int = 0,
+        time_limit: int = 10,
+        memory_block_size_in_kb: int = 1,
+        memory_total_size_in_gb: int = 100,
+        memory_hugetlb: bool = False,
+        memory_oper: str = "write",
+        memory_access_mode: str = "seq",
+        memory_scope: str = "global",
+    ) -> None:
+        block_size = memory_block_size_in_kb * 1024
+        total_mem_size = memory_total_size_in_gb * 1024 * 1024 * 1024
+
+        valid_mem_scope: List[str] = ["global", "local"]
+        valid_mem_operation: List[str] = ["read", "write", "none"]
+        valid_mem_access_mode: List[str] = ["seq", "rnd"]
+        perf_msg: Dict[Any, Any] = {}
+        testcase_name: str = inspect.stack()[1][3]
+        hugetlb: str = "off"
+
+        if memory_scope not in valid_mem_scope:
+            raise LisaException(
+                f"Invalid memory_scope. Valid memory_scope: {valid_mem_scope}"
+            )
+        elif memory_oper not in valid_mem_operation:
```

Comment:
Similar with fileio, use 3 `if`, instead of `elif` on unrelated checks.

Reply:
Done